### PR TITLE
UCS/RCACHE: optimize region merging

### PR DESCRIPTION
**Pull request by alex-mikheev**
@yosefe please take a look

### CODE REVIEW

File: src/ucs/sys/rcache.c [hunk 1]
```diff
@@ -348,9 +328,24 @@ ucs_rcache_check_overlap(ucs_rcache_t *rcache, ucs_pgt_addr_t *start,
          * the next time.
          */
         if (!ucs_test_all_flags(*prot, region->prot)) {
+            /* A slow path because searching /proc/maps in order to
+             * check memory protection is very expensive.
+             *
+             * NOTE: currently ib_md which is the only user of rcache creates
```
Comment (yosefe) [inline src/ucs/sys/rcache.c]: this note should go to ib_md.c.
is this code reached in one of the unit tests?

File: src/ucs/sys/rcache.c [hunk 2]
```diff
@@ -348,9 +328,24 @@ ucs_rcache_check_overlap(ucs_rcache_t *rcache, ucs_pgt_addr_t *start,
...
+             * all regions with the same (r|w) protection so this code is
+             * never reached.
+             */
             mem_prot = UCS_PROFILE_CALL(ucs_get_mem_prot, *start, *end);
-            ucs_assert(ucs_test_all_flags(mem_prot, *prot));
-            if (ucs_test_all_flags(mem_prot, region->prot)) {
+            if (!ucs_test_all_flags(mem_prot, *prot)) {
+                ucs_rcache_region_trace(rcache, region,
+                        "do not merge "UCS_RCACHE_PROT_FMT
```
Comment (yosefe) [inline src/ucs/sys/rcache.c]: indentation

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4358/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4386/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4390/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4392/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4395/ for details (Mellanox internal link).

**Comment by alex-mikheev** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4400/ for details (Mellanox internal link).

**Comment by alex-mikheev** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4466/ for details (Mellanox internal link).

**Comment by miked-mellanox** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4486/ for details (Mellanox internal link).

**Comment by miked-mellanox** [comment]
bot:ornl:retest

**Comment by miked-mellanox** [comment]
Ornl bot not responding.merging for mtt coverage. @MattBBaker please check bot.

**Pull request merged**